hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerID.java

```diff
@@ -46,16 +45,20 @@ public static Codec<ContainerID> getCodec() {
   }
 
   private final long id;
+  private final HddsProtos.ContainerID proto;
+  private final int hash;
```

Comment:
We can add `@Immutable` on the class.  (Only if you update the patch anyway.)

(`import net.jcip.annotations.Immutable;`)